modules/http/src/main/scala/ch/epfl/bluebrain/nexus/commons/http/JsonLdCirceSupport.scala

```diff
@@ -29,24 +29,36 @@ trait JsonLdCirceSupport extends FailFastCirceSupport {
     */
   def sortKeys(json: Json)(implicit keys: OrderedKeys): Json = {
 
-    implicit val _: Ordering[String] = new Ordering[String] {
+    implicit val _: Ordering[Key] = new Ordering[Key] {
       private val middlePos = keys.withPosition("")
 
-      private def position(key: String): Int = keys.withPosition.getOrElse(key, middlePos)
+      private def position(key: Key): Int =
+        keys.withPosition.get(s"*${key.raw}") orElse keys.withPosition.get(key.prefixed) getOrElse (middlePos)
 
-      override def compare(x: String, y: String): Int = {
+      override def compare(x: Key, y: Key): Int = {
         val posX = position(x)
         val posY = position(y)
-        if (posX == middlePos && posY == middlePos) x compareTo y
+        if (posX == middlePos && posY == middlePos) x.raw compareTo y.raw
         else posX compareTo posY
       }
     }
 
-    def canonicalJson(json: Json): Json =
-      json.arrayOrObject[Json](json, arr => Json.fromValues(arr.map(canonicalJson)), obj => sorted(obj).asJson)
-
-    def sorted(jObj: JsonObject): JsonObject =
-      JsonObject.fromIterable(jObj.toVector.sortBy(_._1).map { case (k, v) => k -> canonicalJson(v) })
+    def canonicalJson(json: Json, prefixPath: Option[String] = None): Json =
+      json.arrayOrObject[Json](
+        json,
+        arr => Json.fromValues(arr.map(j => canonicalJson(j, prefixPath))),
+        obj =>
+          JsonObject
+            .fromIterable(
+              obj.toVector
+                .sortBy {
+                  case (k, _) => Key(prefixPath.map(p => s"${p}.$k").getOrElse(k), k)
+                }
+                .map {
+                  case (k, v) => k -> canonicalJson(v, Some(prefixPath.map(p => k).getOrElse(k)))
```

Comment:
isn't ` Some(prefixPath.map(p => k).getOrElse(k)` equivalent to just `Some(k)`? 

or shouldn't it be ` Some(prefixPath.map(p => s"${p}.$k").getOrElse(k)` if we want deeper nesting?

Reply:
Sorry, true